src/components/client/ClientNew.js

```diff
@@ -0,0 +1,231 @@
+                      className='form-control inner-form'
+                      id='inputNumber'
+                      onChange={this.handleChange}
+                      required
+                    />
+                  </div>
+                </div>
+                <div
+                  className='form-group col-sm-12 col-md-3 has-primary'
+                  align='center'>
+                  {this.state.logoLoaded === true ? (
+                    <img
+                      src={this.state.logoPreview}
+                      className='fileinput-new thumbnail img-raised logo'
+                    />
+                  ) : (
+                    <div className='fileinput-new thumbnail img-raised logo-place-holder' />
+                  )}
+                  {this.state.logoLoaded === true ? (
+                    <div>
+                      <button
+                        className='btn btn-round btn-primary client-button'
+                        type='button'>
+                        Change Logo
+                        <input
+                          type='file'
+                          ref='logo'
+                          name='logo'
+                          onClick={this.handleLogoChange}
+                          className='fileInput'
+                        />
+                      </button>
+                      <button
+                        class='btn btn-fab btn-danger btn-round client-button'
+                        type='button'
+                        onClick={this.handleLogoDelete}>
+                        <i class='fa fa-times' />
+                      </button>
+                    </div>
+                  ) : (
+                    <button
+                      className='btn btn-round btn-primary client-button'
+                      type='button'>
+                      Select Logo
+                      <input
+                        type='file'
+                        ref='logo'
+                        name='logo'
+                        onChange={this.handleLogoChange}
+                        className='fileInput'
+                      />
+                    </button>
+                  )}
+                </div>
+              </div>
+              {/* buttons */}
+              <div className='form-row'>
+                <div className=' form-group col-xs-1'>
+                  <Link to='/clients'>
+                    <button type='reset' className='btn btn-danger'>
+                      Cancel
+                    </button>
+                  </Link>
+                </div>
+                <div className='form-group col-xs-1 text-end ml-auto'>
+                  <button type='submit' className='btn btn-success btn-right'>
+                    Add
+                  </button>
+                </div>
+              </div>
+            </form>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default withRouter(AddNewProject);
```

Comment:
I'm so impressed what you did with new project and new client! :smile: can you explain what withRouter is? and why it takes AddNewProject?

Reply:
So this is the typical export default "the name of the class component". But I have a thing called history which sends the user back to the previous page. This explains it better https://medium.com/@AkyunaAkish/understanding-react-router-4-df73a66d96c4 - scroll to withRouter and history.listen
This is inside the POST 'this.props.history.push('/clients');'